lib/datadog/tracing/contrib/aws/configuration/settings.rb

```diff
@@ -25,7 +25,15 @@ class Settings < Contrib::Configuration::Settings
             end
 
             option :service_name do |o|
-              o.default { ENV.fetch(Ext::ENV_SERVICE_NAME, Ext::DEFAULT_PEER_SERVICE_NAME) }
+              o.default do
+                ENV.fetch(Ext::ENV_SERVICE_NAME) do
+                  if Datadog.configuration.tracing.span_attribute_schema == 'v1'
+                    Datadog.configuration.service
+                  else
+                    Ext::DEFAULT_PEER_SERVICE_NAME
+                  end
+                end
```

Comment:
Because this is going to be repeated over and over again, I recommend abstracting this behaviour, something like:
```ruby
def fetch_service_name(env, default)
  ENV.fetch(env) do
    if Datadog.configuration.tracing.span_attribute_schema == 'v1'
      Datadog.configuration.service
    else
      default
    end
  end
end
```

Reply:
> Because this is going to be repeated over and over again, I recommend abstracting this behaviour, something like:
> 
> ```ruby
> def fetch_service_name(env, default)
>   ENV.fetch(env) do
>     if Datadog.configuration.tracing.span_attribute_schema == 'v1'
>       Datadog.configuration.service
>     else
>       default
>     end
>   end
> end
> ```

I am expecting this pattern would be repeated in multiple integrations, I recommend extract and abstract the stateless function to be utilities being reusable.